### PR TITLE
update CC provider text when using CC iteration

### PR DESCRIPTION
**Pull request by jenniemc**
## Description
Update CC provider preference page title and main content when using CC iteration flag. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#29640

## Testing done
  Feature: VAOS CommunityCareProviderSelectionPage

    Scenario: should not display closest city question when using iterations toggle
      Given a user with two supported sites
        And the CC iterations toggle is on
        And type of care is selected
      When the page is displayed
      Then the heading will display type of provider requested
        And the closest city/state question is not shown

## Screenshots
CC iteration flag on
image
<hr>

using CC iteration while expanding the provider list
image
<hr>

CC iteration flag off - multi parent site
image

<hr/>
CC iteration flag off - single parent site

image

## Acceptance criteria
- [ ] content matches [copy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/cc-provider-preference.md#iterations-toggle) when using CC iteration flag 
- [ ] Update the h1 to show type of care provider requested

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by jbalboni** [comment]
@jenniemc These changes look good to me at first glance, though it looks like there's a failing test.

**Comment by jenniemc** [comment]
@jbalboni, that's really strange. I re-ran the build and it's now passed.

File: src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx [hunk 1]
```diff
@@ -697,9 +699,18 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
...
 
-    // Then the closest city/state question is not shown
+    // Then the heading will display type of provider requested
+    expect(
+      await screen.findByRole('heading', {
+        level: 1,
+        name: /Request a Primary care provider/i,
+      }),
+    ).to.be.ok;
+    expect(screen.getByText(/We’ll call you to confirm your provider choice/))
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx:710]: It looks like this fails sometimes (it happened in Jenkins and I saw it happen once locally). Might be worth doing `expect(await screen.findByText(/We’ll call you to confirm your provider choice/)).to.be.ok;`

File: src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx [hunk 2]
```diff
@@ -689,6 +690,7 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
         { id: '984', address: { city: 'Belgrade', state: 'MT' } },
       ],
     });
+    await setTypeOfCare(store, /primary care/i);
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx:693]: I don't think you need this, the type of care is set in `setCommunityCareFlow`

**Comment by jbalboni** [comment]
I get that extra label text when the CC iterations flag is off and there's only one parent site (i.e. we don't show the question):

Screen Shot 2021-09-15 at 9 55 47 AM

**Comment by jenniemc** [comment]
> I get that extra label text when the CC iterations flag is off and there's only one parent site (i.e. we don't show the question):
> 
> Screen Shot 2021-09-15 at 9 55 47 AM

@jbalboni  great catch! I didn't even think about scenario of single parent site. Turns out the original `new-appointment/redux/reducer.js` file accounts for the single and multi parent site. Reverted back reducer file.

**Review by vbahinwillit — COMMENTED** [review]
LGTM

**Review by jbalboni — APPROVED** [review]

**Pull request merged**